memory-viz/src/tests/cli.spec.tsx

```diff
@@ -19,94 +20,131 @@ const input = JSON.stringify(
-            (err) => {
-                if (err) throw err;
-                const svgFilePath = path.resolve(
-                    process.cwd(),
-                    path.basename(filePath.replace(".json", ".svg"))
-                );
-                const fileContent = fs.readFileSync(svgFilePath, "utf8");
-                expect(fileContent).toMatchSnapshot();
-                fs.unlinkSync(svgFilePath);
-                done();
-            }
-        );
+        const child = spawn(command, args, { shell: true });
+
+        child.stdin.write(input);
+        child.stdin.end();
+
+        let output = "";
+        child.stdout.on("data", (data) => {
+            output += data.toString();
+        });
+
+        child.on("close", (err) => {
+            if (err) throw err;
+            expect(output).toMatchSnapshot();
+            done();
+        });
     });
 
-    it("produces consistent svg when provided height and width options", (done) => {
-        fs.writeFileSync(filePath, input);
+    it("produces consistent svg when provided stdin and output", (done) => {
+        const command = "memory-viz";
+        const args = [`--output=${outputPath}`, "--roughjs-config seed=1234"];
 
-        exec(
-            `memory-viz ${filePath} --height=700 width=1200 --roughjs-config seed=12345`,
-            (err) => {
-                if (err) throw err;
-                const svgFilePath = path.resolve(
-                    process.cwd(),
-                    path.basename(filePath.replace(".json", ".svg"))
-                );
-                const fileContent = fs.readFileSync(svgFilePath, "utf8");
-                expect(fileContent).toMatchSnapshot();
-                fs.unlinkSync(svgFilePath);
-                done();
-            }
-        );
+        const child = spawn(command, args, { shell: true });
+
+        child.stdin.write(input);
+        child.stdin.end();
+
+        child.on("close", (err) => {
+            if (err) throw err;
+
+            const svgFilePath = getSVGPath(true);
+            const fileContent = fs.readFileSync(svgFilePath, "utf8");
+            expect(fileContent).toMatchSnapshot();
+            fs.unlinkSync(svgFilePath);
+            done();
+        });
     });
 
-    it("produces consistent svg when provided a variety of rough-config options", (done) => {
+    it("logs a warning when both --stdout and --output are specified", (done) => {
         fs.writeFileSync(filePath, input);
 
         exec(
-            `memory-viz ${filePath} --roughjs-config seed=1234,fill=red,fillStyle=solid`,
-            (err) => {
+            `memory-viz ${filePath} --stdout --output=${outputPath}`,
+            (err, stdout, stderr) => {
                 if (err) throw err;
-                const svgFilePath = path.resolve(
-                    process.cwd(),
-                    path.basename(filePath.replace(".json", ".svg"))
+                const output = stdout + stderr;
```

Comment:
you should not need to combine stdout and stderr here, the error message is only printed to one or the other